components/camel-pulsar/src/main/java/org.apache.camel.component.pulsar/PulsarProducer.java

```diff
@@ -60,17 +63,26 @@ public void process(final Exchange exchange) throws Exception {
     private synchronized void createProducer() throws org.apache.pulsar.client.api.PulsarClientException {
         if (producer == null) {
             final String topic = pulsarEndpoint.getTopic();
-            String producerName;
-            if (pulsarEndpoint.getConfiguration().getProducerName() == null) {
+            PulsarEndpointConfiguration configuration = pulsarEndpoint.getConfiguration();
+            String producerName = configuration.getProducerName();
+            if (producerName == null) {
                 producerName = topic + "-" + Thread.currentThread().getId();
-            } else {
-                producerName = pulsarEndpoint.getConfiguration().getProducerName();
             }
             final ProducerBuilder<byte[]> producerBuilder = pulsarEndpoint
                     .getPulsarClient()
                     .newProducer()
                     .producerName(producerName)
-                    .topic(topic);
+                    .topic(topic)
+                    .sendTimeout(configuration.getSendTimeoutMs(), TimeUnit.MILLISECONDS)
+                    .blockIfQueueFull(configuration.isBlockIfQueueFull())
+                    .maxPendingMessages(configuration.getMaxPendingMessages())
+                    .maxPendingMessagesAcrossPartitions(configuration.getMaxPendingMessagesAcrossPartitions())
+                    .batchingMaxPublishDelay(configuration.getBatchingMaxPublishDelayMicros(), TimeUnit.MICROSECONDS)
+                    .batchingMaxMessages(configuration.getMaxPendingMessages())
+                    .enableBatching(configuration.isBatchingEnabled());
+            if (configuration.getInitialSequenceId() != null) {
+                producerBuilder.initialSequenceId(configuration.getInitialSequenceId());
```

Comment:
It could be set to less than one here with the docs saying minimum value is 1. This would simplify the rule for users setting the config.